autopush/utils.py

```diff
@@ -108,13 +108,39 @@ def fix_padding(string):
     return string
 
 
+def decipher_public_key(key_data):
+    """A public key may come in several flavors. Attempt to extract the
+    valid key bits from keys doing minimal validation checks.
+
+    This is mostly a result of libs like WebCrypto prefixing data to "raw"
+    keys, and the ecdsa library not really providing helpful errors.
+
+    :param key_data: the raw-ish key we're going to try and process
+    :returns: the raw key data.
+    :raises: ValueError for unknown or poorly formatted keys.
+
+    """
+    # key data is actually a raw coordinate pair
+    key_len = len(key_data)
+    if key_len == 64:
+        return key_data
+    # Key format is "raw"
+    if key_len == 65 and key_data[0] == '\x04':
+        return key_data[-64:]
+    # key format is "spki"
+    if key_len == 88 and key_data[:3] == '0V0':
```

Comment:
Sorry for the swoop-and-:poop: here, but how do we know the last 3 should be `0V0`? I'm asking because I'm trying to mirror [this code (and test code) in the push dashboard](https://github.com/mozilla-services/push-dev-dashboard/compare/add-vapid-key-to-apps-37?expand=1):
```
def _gen_keys():
    signing_key = ecdsa.SigningKey.generate(curve=ecdsa.NIST256p)
    verifying_key = signing_key.get_verifying_key()
    vapid_key = urlsafe_b64encode(verifying_key.to_string())
    return (signing_key, verifying_key, vapid_key)

...

    def test_extract_public_key_expected(self):
        signing_key, verifying_key, vapid_key = _gen_keys()
        self.assertEqual(vapid_key, extract_public_key(vapid_key)) 
```
But I'm getting:
```
ValueError: Unknown public key format specified
```
Because `key_data[:3]` is different each time? Is it because `utils.fix_padding` here always makes trailing characters into `=` character?